### PR TITLE
Fix/unify autocorr interface

### PR DESCRIPTION
**Pull request by MFreidank**
Following previous discussion in issue #141 this pull request fixes inconsistencies between autocorrelation computation in `az.stats.diagnostics.autocorr` (previously `az.stats.diagnostics._autocorr`) and `az.stats.plots.autocorrplot`. The notions of autocorrelation computation differed previously, in this pull request we settle for the (recently thoroughly discussed) notion as (recently, in February 2018) implemented in `az.stats.diagnostics.autocorr`, `pymc3.stats._autocorr` and `stan::math::autocorrelation`. 

As proof of concept and to make differences between old and new interface visible, consider the following plots:

New:
autocorrplot_new

Previous:
autocorrplot_old

### CODE REVIEW

File: arviz/plots/autocorrplot.py [hunk 1]
```diff
@@ -56,13 +57,10 @@ def autocorrplot(posterior, var_names=None, max_lag=100, symmetric_plot=False, c
     for (var_name, selection, x), ax in zip(plotters, axes.flatten()):
         if combined:
             x = x.flatten()
-        y = x - x.mean()
-        y = np.correlate(y, y, mode=2)
-        y = y / np.abs(y).max()
-        midpoint = len(y) // 2
+        y = autocorr(x)
         ax.vlines(x=np.arange(min_lag, max_lag),
-                  ymin=np.zeros(max_lag - min_lag),
```
Comment (ColCarroll) [inline arviz/plots/autocorrplot.py]: this was allowing for `symmetric_plot=True`, though I am not sure I see the use of that option any more.

Comment (ColCarroll) [reply arviz/plots/autocorrplot.py]: Specifically, I think this is fine, but either here or in a followup PR we should remove the `symmetric_plot` option.

**Comment by ColCarroll** [comment]
I restarted the hpd test. it should probably be run with a fixed random seed since it fails very occasionally.

File: arviz/plots/autocorrplot.py (hunk 1 above)
Comment (MFreidank) [reply arviz/plots/autocorrplot.py]: I removed the option. `autocorr(x)` already removes symmetry, so we would have to reintroduce it after that computation (by essentially appending a reversed copy of `y` to itself). I think this is a rare, specialized usecase that I have not seen used in actual plots too much.

**Comment by MFreidank** [comment]
@ColCarroll I also introduced capping `y_lim` at `(-1, 1)` - I think that makes the most sense as these are the natural bounds of autocorrelation data. (and in the rare case where someone would need to adjust these manually, it is still possible as we return the axis after the fact)

**Comment by MFreidank** [comment]
Shouldn't the last line be  `return axes` instead of `return ax`?

**Comment by ColCarroll** [comment]
Wow, yeah, that code is a little nasty (it deals with the edge case that there are 0 plots, so `ax` doesn't exist after the loop).  

I might also change the end to 

```python
if axes.size > 0:
    axes[0, 0].set_xlim(...)
    axes[0, 0].set_ylim(...)
```

Note that this is ok because of the earlier `np.atleast_2d`, and we passed `sharex=True, sharey=True` to the `plt.subplots` command.

**Comment by MFreidank** [comment]
@ColCarroll I refactored that as outlined by you (I think it conveys the intention slightly more clearly) and we now return all axes instead of just the last one processed in the `for` loop. 

Plots now look like this:
autocorrplot_new

File: arviz/tests/test_plots.py [hunk 2]
```diff
@@ -40,7 +40,7 @@ def test_posteriorplot(self):
 
     def test_autocorrplot(self):
         for obj in (self.short_trace, self.fit):
-            assert autocorrplot(obj).get_geometry() == (6, 6, 36)
+            assert autocorrplot(obj)[0, 0].get_geometry() == (6, 6, 36)
```
Comment (ColCarroll) [inline arviz/tests/test_plots.py]: elsewhere we're using `assert autocorrplot(obj).shape == (6, 6)`

Comment (MFreidank) [reply arviz/tests/test_plots.py]: yeah, I like that better. Changed it to that now.

**Comment by MFreidank** [comment]
Travis passed. Could this be merged?

**Comment by ColCarroll** [comment]
Thanks again for all this careful work!

**Pull request merged**